CMakeLists.txt

```diff
@@ -56,6 +56,53 @@ if(BUILD_WIN32
     set(BUILD_WIN TRUE)
 endif()
 
+SET(ZT_ARCHITECTURE 999)
+IF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "amd64")
+    SET(ZT_ARCHITECTURE 2)
+    SET(ZT_USE_X64_ASM_SALSA 1)
+    SET(ZT_USE_X64_ASM_ED25519 1)
+    SET(ZT_SSO_SUPPORTED 1)
+    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse -msse2")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse -msse2")
+ELSEIF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "powerpc64le")
+    SET(ZT_ARCHITECTURE 8)
+    ADD_DEFINITIONS(-DZT_NO_TYPE_PUNNING)
+ELSEIF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "powerpc")
+    SET(ZT_ARCHITECTURE 8)
+    ADD_DEFINITIONS(-DZT_NO_TYPE_PUNNING -DZT_NO_CAPABILITIES)
+ELSEIF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "ppc64el")
+    SET(ZT_ARCHITECTURE 8)
+ELSEIF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "e2k")
+    SET(ZT_ARCHITECTURE 2)
+ELSEIF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "i386" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "i486" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "i586" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "i686")
+    SET(ZT_ARCHITECTURE 1)
+    SET(ZT_SSO_SUPPORTED 1)
+ELSEIF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armel" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armhf" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv6" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv6l" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv6zk" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv7" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv7l" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv7hl" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv7ve")
+    SET(ZT_ARCHITECTURE 3)
+    SET(ZT_USE_ARM32_NEON_ASM_CRYPTO 1)
+    ADD_DEFINITIONS(-DZT_NO_TYPE_PUNNING)
+ELSEIF (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
+    SET(ZT_ARCHITECTURE 4)
+    SET(ZT_SSO_SUPPORTED 1)
+    ADD_DEFINITIONS(-DZT_NO_TYPE_PUNNING -DZT_ARCH_ARM_HAS_NEON -march=armv8-a+crypto -mtune=generic -mstrict-align)
```

Comment:
This is wrong. The Armv8A crypto extension is optional and not all CPUs support it. For example Raspberry Pi 4 doesn't, it will compile but crash with `illegal instruction`.